src/chalk_common.nim

```diff
@@ -279,10 +279,9 @@ type
   DockerManifestType* = enum
     list, image, config, layer
 
-  DockerPlatform* = tuple[
-    os:           string,
-    architecture: string,
-  ]
+  DockerPlatform* = object
+    os*:           string
+    architecture*: string
```

Comment:
this was explicitly done as a tuple so that all comparisons are done correctly by nim. does nim do the right thing for `==` on objects as well?